backend/apps/ecommerce/resolvers.py

```diff
@@ -38,9 +38,14 @@ def resolve_user_orders(self, info):
     def resolve_all_user_orders(self, info):
         return Order.objects.all()
 
-    @staff_member_required
-    def resolve_all_shop_orders(self, info):
-        return Order.objects.filter(product__shop_item=True)
+    def resolve_all_shop_orders(self, info, limit, offset):
+        # Apply pagination if limit and offset are provided
+        orders = Order.objects.filter(product__shop_item=True).order_by('delivered_product', 'payment_status', 'timestamp')
+        if offset is not None:
+            orders = orders[offset:]
+        if limit is not None:
```

Comment:
Can you instead of this add a max of 300 to limit. So if I set it ti 400 it caps at 300 instead. This is to prevent someone from unknowingly crashing the thing later. Also add a short comment on why it is capped at 300 as it won't be obvious to anyone not extreamly familiar with our system